#!/usr/bin/env python
PACKAGE = "ghost"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Car limits
gen.add("v_limit", double_t, 0, "Maximum velocity", 2.0, 0.01, 15.0)
gen.add("ax_accel_limit", double_t, 0, "Maximum longitudinal acceleration forward", 2.0, 0.01, 15.0)
gen.add("ax_braking_limit", double_t, 0, "Maximum longitudinal acceleration braking", 0.5, 0.01, 5.0)
gen.add("ay_limit", double_t, 0, "Maximum lateral acceleration", 1.5, 0.01, 15.0)

# Steering PID controller parameters
gen.add("lookahead_dist", double_t, 0, "Distance to look ahead for heading error", 1.0, 0.0, 3.0)
gen.add("K_psi", double_t, 0, "Steering controller proportional coefficient", 0.3, 0.00, 5.0)
gen.add("Ti_psi", double_t, 0, "Steering controller integral coefficient", 5.0, 0.00, 50.0)
gen.add("Td_psi", double_t, 0, "Steering controller derivative coefficient", 0.0, 0.00, 10.0)

# Velocity PID controller parameters
gen.add("K_v", double_t, 0, "Velicity controller proportional coefficient", 1.0, 0.0, 2.0)
gen.add("Ti_v", double_t, 0, "Velicity controller integral coefficient", 10.0, 0.0, 50.0)
gen.add("Td_v", double_t, 0, "Velicity controller derivative coefficient", 0.0, 0.0, 10.0)

exit(gen.generate(PACKAGE, "ghost", "Controller"))
